src/NServiceBus.AcceptanceTesting/ScenarioWithContext.cs

```diff
@@ -97,11 +99,67 @@ public IScenarioWithEndpointBehavior<TContext> Done(Func<TContext, bool> func)
 
     public IScenarioWithEndpointBehavior<TContext> Done(Func<TContext, Task<bool>> func)
     {
-        done = c => func((TContext)c);
+        if (doneTask is not null)
+        {
+            throw new InvalidOperationException("Done condition has already been defined.");
+        }
+        doneTask = Task.Run(async () =>
```

Comment:
This is more for backward compatibility and this complexity could be removed when things are migrated